src/nn/Execution.ts

```diff
@@ -0,0 +1,51 @@
+import { assert, TypedArray, createTensor, validateTypedArray } from './utils';
+import { Input } from './Input';
+import { Output } from './Output';
+import { Compilation } from './Compilation';
+import { ExecutionContext } from './ExecutionContext';
+
+import * as tf from '@tensorflow/tfjs-core'
+
+export class Execution {
+  private compilation_: Compilation;
+  private inputTensors_: Map<Input, tf.Tensor> = new Map();
+  private outputBuffers_: Map<Output, TypedArray> = new Map();
+
+  constructor(compilation: Compilation) {
+    this.compilation_ = compilation;
+  }
+
+  setInput(name: string, data: TypedArray): void {
+    assert(typeof name === 'string' &&
+        this.compilation_.model_.inputs_.has(name), 'The name parameter is invalid.');
+    const input = this.compilation_.model_.inputs_.get(name);
+    validateTypedArray(data, input.desc);
+    this.inputTensors_.set(input, createTensor(input.desc, data));
+  }
+
+  setOutput(name: string, data: TypedArray): void {
```

Comment:
curious why setting value for output? is this for holding final prediction results?
should this be done automatically? since the model can infer the output shape, it is an extra step for user to provide the storage?

Reply:
> curious why setting value for output? is this for holding final prediction results?

Yes. It sets the buffer to hold the results.

> should this be done automatically? since the model can infer the output shape, it is an extra step for user to provide the storage?

This implementation follows the current [API spec](https://webmachinelearning.github.io/webnn/#dom-execution-setoutput). 

But you raised a good point. I guess the reason of providing the storage is to facilitate the buffer reuse. If you see this is an issue, please feel free to file an API bug to [WebNN issues](https://github.com/webmachinelearning/webnn/issues).

Reply:
Sounds good, I think I do have couple suggestions related to the execution.

Reply:
Will follow up this in the [Execution API issue](https://github.com/webmachinelearning/webnn/issues/87) of webnn repo.